internal/page/donor/your_preferred_language.go

```diff
@@ -0,0 +1,49 @@
+package donor
+
+import (
+	"net/http"
+
+	"github.com/ministryofjustice/opg-go-common/template"
+	"github.com/ministryofjustice/opg-modernising-lpa/internal/actor"
+	"github.com/ministryofjustice/opg-modernising-lpa/internal/form"
+	"github.com/ministryofjustice/opg-modernising-lpa/internal/localize"
+	"github.com/ministryofjustice/opg-modernising-lpa/internal/page"
+	"github.com/ministryofjustice/opg-modernising-lpa/internal/validation"
+)
+
+type yourPreferredLanguageData struct {
+	App        page.AppData
+	Errors     validation.List
+	Form       *form.LanguagePreferenceForm
+	Options    localize.LangOptions
+	FieldNames form.Names
+}
+
+func YourPreferredLanguage(tmpl template.Template, donorStore DonorStore) Handler {
+	return func(appData page.AppData, w http.ResponseWriter, r *http.Request, donor *actor.DonorProvidedDetails) error {
+		data := &yourPreferredLanguageData{
+			App: appData,
+			Form: &form.LanguagePreferenceForm{
+				Preference: donor.ContactLanguagePreference,
+			},
+			Options:    localize.LangValues,
+			FieldNames: form.FieldNames,
```

Comment:
maybe it would be better to have `FieldName` and just pass in the one needed, if we carry on this pattern